Python/other/minimize_the_jump.py

```diff
@@ -0,0 +1,40 @@
+#Find Minimum number of jumps to reach end
+def min_jump(arr, n):
+    if n <= 1: 
+        return 0 #return zero values if the array less than one element
+    
+    #assigning the value 0    
+    left = 0
+    right = arr[0]
+    count = 1
+
+    while right < n-1:
+        count += 1 #incresing the value 
+        jump = max(i + arr[i] for i in range(left, right + 1)) #by this checking all the arr and finding min values 
+        left, right = right, jump #swaping the values
+    return count
+
+arr = list(map(int, input("Enter the elements: ").split()))
+size = len(arr)
+print('Minimum number of jumps to reach',
+      'end is', min_jump(arr, size))
+
+'''
+Time Complexity : O(log(n))
```

Comment:
Explain how you concluded the written Time Complexity

Reply:
```suggestion
Time Complexity : O(log(n))
```

![IMG-20210408-WA0002](https://user-images.githubusercontent.com/66599363/113989089-b7f3d680-986d-11eb-8c1c-e98a672cf682.jpg)

It's contain iteration so the formula of the time complexity is not be calulated like recursion. but loop kitni baar chalega, is cae me wo n , matlab puri leght of array se lam aur 1 se jyada baar chal raha hai lsliye ye O(log n ) hoga mere hisaab se to

Reply:
@siddharth25pandey  any problem in my solution please tell me

Reply:
@soumik2012 As you are iterating your outer while loop as **while right < n-1**, its sure that it runs less than **n iteration** but doesn't mean the Time Complexity will be **O(logn)**. It will be O(logn) when we can write like this:
`T(n)=T(n/2)+O(c)` where c is a constant. Ex: Binary Search Algorithm.
Hence, the loop is Iterating for n-1 times means O(n-1) is almost equal to  O(n) in general
Therefore change the Time Complexity to **O(n)**